Update got, fix tests, and update .travis.yml

sonicdoe opened the pull request.

sonicdoe committed:
Fix linting

SamVerschueren commented on package.json:
```diff
@@ -36,9 +36,9 @@
     "got": "^6.3.0"
   },
   "devDependencies": {
-    "ava": "*",
```
This is on purpose, same for `XO`. I do this together with @sindresorhus and some others to make sure that we always test our own projects against the latest versions. This way we can detect bugs from those libraries early on.

sonicdoe replied:
Ah, thanks for the clarification.

SamVerschueren reviewed (changes requested):
Can you also change `.travis.yml` to test against `6` and `4` instead of `5` and `4`?

sonicdoe committed:
Update .travis.yml

SamVerschueren commented:
Have to look into the Travis tests. Not sure why it fails, maybe because it runs on Travis and they are throttled. Does it work locally for you?

sonicdoe commented:
At first I thought it was caused by rate limiting, too. However, the tests didn’t work for me locally as well. I have tried to fix the tests in #2.

sonicdoe committed:
Update got to v7

sonicdoe committed:
Fix tests

sonicdoe commented:
Since got v7 was released today, I have updated this pull request. The tests are now working locally for me (as long as I have a valid `TRAVIS_TOKEN` set). @SamVerschueren Could you take a look again?

SamVerschueren commented:
Do you have any idea why the tests are failing? Hard to tell on mobile :)

SamVerschueren commented:
Sorry, didn't read your last comment. Will have a look tomorrow.

sonicdoe committed:
Fix TRAVIS_TOKEN being set to `undefined`

Assigning a property on `process.env` will implicitly convert the value to a string, see https://nodejs.org/dist/latest-v6.x/docs/api/process.html#process_process_env. This meant that if TRAVIS_TOKEN was unset it was re-set to the string `undefined`.

sonicdoe commented:
Well, at least you prompted me to look into this in more detail :)

The actual issue was that if `TRAVIS_TOKEN` was not set when running the tests, the “global token option” test would re-set it using `process.env.TRAVIS_TOKEN = token`. However, [assigning a property on `process.env` will implicitly convert the value to a string](https://nodejs.org/dist/latest-v6.x/docs/api/process.html#process_process_env) which meant that `process.env.TRAVIS_TOKEN` was literally assigned the string `undefined`.

I have now fixed this by simply setting `process.env.TRAVIS_TOKEN` to an empty string in case of `TRAVIS_TOKEN` being undefined.
